### PR TITLE
Refactor rabit tests

### PR DESCRIPTION
**Pull request by trivialfis**
* Merge RABIT google tests into XGBoost.
* Run standalone tests in CTest.
* Run tests on Jenkins.
* Simplify CMakeLists.txt, fixing #6092 .

### CODE REVIEW

**Comment by trivialfis** [comment]
Somehow there are duplicated tests defined in cc files and cpp files.

**Comment by trivialfis** [comment]
@hcho3 I will do a drive by fix for #6092 .

File: rabit/CMakeLists.txt [hunk 1]
```diff
@@ -94,89 +63,21 @@ if (RABIT_BUILD_TESTS)
     add_executable(${test} test/${test}.cc)
     target_link_libraries(${test} rabit_mock_static)
     set_target_properties(${test}  PROPERTIES CXX_STANDARD 11 CXX_STANDARD_REQUIRED ON)
```
Comment (hcho3) [inline rabit/CMakeLists.txt:65]:
```suggestion
    set_target_properties(${test}  PROPERTIES CXX_STANDARD 14 CXX_STANDARD_REQUIRED ON)
```

**Review by hcho3 — CHANGES_REQUESTED** [review]
I did a first round of review.

You also need to change line 287 of the root CMakeLists.txt: We should install `rabit` along with `objxgboost` when `BUILD_STATIC_LIB` is on. So add it to the variable `INSTALL_TARGETS`. The reason is that Rabit is a dependency of `xgboost`, and when you install a static lib, you need to install all of its dependencies.

**Comment by hcho3** [comment]
I'll do another round of review tomorrow, as I am about to go to bed now.

**Comment by trivialfis** [comment]
@hcho3 Thanks for the explanation.  Let me do some tests, I think it's possible to link multiple static libraries into one.

**Comment by hcho3** [comment]
> I think it's possible to link multiple static libraries into one.

You can turn rabit into an object library, so that librabit.a won't need to be installed. Even here, we still need to declare rabit in the export set of INSTALL(TARGETS) command.

**Comment by trivialfis** [comment]
Got it, so I need to restore the installation logic.

**Comment by hcho3** [comment]
No, you can just use the installation logic in the **root** CMakeLists.txt. No need to maintain installation logic in two places.

**Comment by trivialfis** [comment]
@hcho3 Thanks again.  Modified the script to install it along with objxgboost when building static library.

**Review by hcho3 — APPROVED** [review]

**Pull request merged**